.github/workflows/bump-version.yml

```diff
@@ -1,44 +1,106 @@
+name: Bump Version
+
 on:
-    pull_request:
-      branches:
-        - main
-      types: [closed]
-  
+  pull_request:
+    types: [closed]
+    branches: [main]
+
 jobs:
-    bump-version:
-      runs-on: ubuntu-latest
-  
-      env:
-        GH_PUSH_TOKEN: ${{ secrets.GH_PUSH_TOKEN }}
-  
-      steps:
-  
-        - name: Checkout
-          uses: actions/checkout@v4
-          if: env.GH_PUSH_TOKEN != ''
-          with:
-            token: ${{ env.GH_PUSH_TOKEN }}
-  
-        - name: configure git
-          if: env.GH_PUSH_TOKEN != ''
-          run: |
-            git remote set-url origin https://itsalaidbacklife@github.com/cuttle-cards/cuttle
-            git config --global user.name "Cuttle Continuous Integration"
-            git config --global user.email "itsalaidbacklife@gmail.com"
-  
-  
-        - name: bump patch version
-          if: contains(github.event.pull_request.labels.*.name, 'version-patch') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:patch
-  
-        - name: bump minor version
-          if: contains(github.event.pull_request.labels.*.name, 'version-minor') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:minor
-  
-        - name: bump major version
-          if: contains(github.event.pull_request.labels.*.name, 'version-major') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:major
-  
-        - name: push
-          if: env.GH_PUSH_TOKEN != ''
-          run: git push origin main
+  check-token:
```

Comment:
2. **Check Token:** The `check-token` job runs first. It verifies if the `GH_PUSH_TOKEN` secret is set. If it's not, the job fails and the workflow terminates. If the secret is set, the job outputs the token for subsequent jobs.